AppDB/datastore_server.py

```diff
@@ -3653,10 +3655,30 @@ def __extract_entities_from_composite_indexes(self, query, index_result):
       if definition.ancestor() == 1:
         ancestor = tokens.pop(0)[:-1]
       distinct_str = ""
+      value_index = 0
       for def_prop in definition.property_list():
-        value = tokens.pop(0) 
+        # If the value contained the separator, try to recover the value.
+        if len(tokens[:-1]) != len(definition.property_list()):
```

Comment:
I don't think it's possible if things are working as intended. The AppServer stops the query if there's not an index defined that matches the query. It shouldn't use a composite index if there are more values in the index than properties in the query.

I changed it to '>', though, since I think it conveys the issue more clearly.